app/src/components/main/analytics/events/common/testItemPages/modalEventsCreators.js

```diff
@@ -46,50 +48,101 @@ const getOpenModalEventCreator = (place) => (defectFromTIGroup, actionPlace = ''
   place: `${place}${actionPlace && `#${actionPlace}`}`,
   condition: DEFECT_FROM_TI_GROUP_MAP[defectFromTIGroup] || 'bulk',
 });
-const getClickOnApplyBtnEventCreator = (place) => (
+
+const getTypePrefix = (prefix, issue) => `${prefix}${issue ? `#${issue}` : ''}`;
+
+const getElementName = (issueActionType) => (issueActionType ? 'apply_and_continue' : 'apply');
+
+const getDefectTypesAnalyticsData = (issueType) => {
+  const defaultDefectTypeNumber = '001';
+  const unselectedIssueType = 'unselected';
+
+  return issueType
+    ? DEFECT_TYPES_LOCATORS_TO_DEFECT_TYPES[issueType] ??
+        `custom_${
+          DEFECT_TYPES_LOCATORS_TO_DEFECT_TYPES[issueType?.slice(0, 2) + defaultDefectTypeNumber]
+        }`
+    : unselectedIssueType;
+};
+
+const getClickOnApplyEventCreator = (place) => (
   defectFromTIGroup,
   hasSuggestions,
   status,
-  switcher,
-  iconName,
+  issueType,
+  itemDataIssueType,
+  issueActionType,
+  suggestedItems,
 ) => {
   const basicEventParameters = getBasicClickEventParametersMakeDecisionCreator(
     place,
     defectFromTIGroup,
   );
-  const isBulkOperation = basicEventParameters.condition === 'bulk';
-  if (!isBulkOperation) {
-    basicEventParameters.type = hasSuggestions ? 'with_ml' : 'without_ml';
-  }
+
+  basicEventParameters.type = hasSuggestions
+    ? getTypePrefix('with_ml', issueActionType)
+    : getTypePrefix('without_ml', issueActionType);
```

Comment:
Can be simplified
```
basicEventParameters.type = getTypePrefix(hasSuggestions ? 'with_ml' : 'without_ml', issueActionType); 
```
or the function `getTypePrefix` can be removed at all.

Reply:
Done.